### ISSUE TITLE
Replace mocking in tests with builders

### ISSUE THREAD

**Issue by ShivangMishra** [eclipse/jkube#2316]
### Component

None

### Description
![image](https://github.com/eclipse/jkube/assets/35092323/1410f371-60a3-45c0-bf5c-e151c98ff3f0)
Currently, in BaseGeneratorTest, JavaExecGeneratorTest, WebAppGeneratorTest, etc. we use mockito to mock the generator which we want to test. We can replace the mocks by building the generator and the java project using the corresponding JavaProject and GeneratorContext builders.

### Expected Behavior
This is an example of how the tests should be written -
![image](https://github.com/eclipse/jkube/assets/35092323/1867ae66-e2e2-4496-9037-41dcccce92a7)

### Acceptance Criteria
- [ ] AC1

**Comment by ShivangMishra** [comment]
@rohanKanojia does this issue look okay? 
I'll create a PR for this one after GSoC.

**Comment by rohanKanojia** [comment]
Looks okay to me. Could you also list the test classes that would require changes in order to fix this issue?

I think someone else can also pick it up if we explain the issue clearly.

**Comment by ShivangMishra** [comment]
@rohanKanojia lots of test classes use mock. I'll make a google sheet and will add it in the description. Would that be okay?

@ShantKhatri if you're interested, please go through this issue and make a list... You can start by making similar changes in the generators. Make a draft PR once this much is done.

**Comment by rohanKanojia** [comment]
@ShivangMishra : Yeah, sounds good. Just make sure permissions are open for anyone.

**Comment by sunix** [comment]
I am not sure to understand the value, it makes the test "bigger" from what I see

**Comment by ShivangMishra** [comment]
> I am not sure to understand the value, it makes the test "bigger" from what I see

Instead of using mocks, we should test on real objects. As far as I understand, we should use mocks only when we cannot directly test on real objects(maybe because the implementation is not complete).

This will prevent complications like we got that Null Pointer Exception earlier.

**Comment by ShantKhatri** [comment]
@rohanKanojia @ShivangMishra I have created list of files which includes mock testing. Here is the google sheet: https://docs.google.com/spreadsheets/d/14bcELvvAVEx5bI30GNgktjqVe6zrx562pRNjjVeI7aA/edit?usp=sharing
Could you please guide me more on this? so I can start work on it.

**Comment by ShantKhatri** [comment]
> Hi @ShantKhatri, Are you working on this?

I was waiting for replay from @rohanKanojia and @ShivangMishra that which from above sheet mocking should we have to replace with generator tests?

**Comment by ShantKhatri** [comment]
@rohanKanojia @manusa Do I need to create a separate issue for each file on which I'm working, or can I create a PR for this (https://github.com/eclipse/jkube/issues/2316) issue only? In that case, there would be multiple PRs linked with a single issue.

**Comment by manusa** [comment]
I'd start just with one of the files so that you get feedback on your approach.

**Comment by ShantKhatri** [comment]
> I'd start just with one of the files so that you get feedback on your approach.

@manusa Could you please guide me with approach?
